### ISSUE TITLE
Regular expressions containing "\u" throw an error

### ISSUE THREAD

**Issue by hansonw** [atom/superstring#52]
Attempting to search for a regular expression containing "\u", even they're escaped as "\\\u", causes errors to be thrown. Easy repro from Atom 1.25:

```
te = atom.workspace.getActiveTextEditor();
buf = te.getBuffer();
buf.findAll(/\\u/)
```

Or if you're using `vim-mode-plus`, try searching for `\\u` in regex mode.

The result is the exception `Uncaught (in promise) Error: PCRE does not support \L, \l, \N{name}, \U, or \u`.

I suspect the culprit is https://github.com/atom/superstring/blob/master/src/core/regex.cc#L18, which attempts to handle `\u` sequences but does not handle the case where the `\` is escaped (`\\u`).

**Comment by 50Wliu** [comment]
I think this is the cause of atom/find-and-replace#999.

**Comment by lee-dohm** [comment]
@hansonw Awesome, thanks! We'll take a look at it when we get the chance but if you have time to toss together a PR for it, we'd appreciate it ✨

**Comment by lee-dohm** [comment]
Double check #999 after a fix is added for this. /cc @smashwilson

**Comment by smashwilson** [comment]
Fixed in #63 and shipped to :atom: in atom/atom@553dec3e. 🤘 

<img width="414" alt="screen shot 2018-03-29 at 1 17 50 pm" src="https://user-images.githubusercontent.com/17565/38103149-99a64d84-3353-11e8-8772-8db79d3addbe.png">
